Describe version attribute in the grammar (#6)
KhronosGroup/SPIRV-Guide: chapters/creating_extension.md

  },
```

Entries should be ordered by their enumerant value to make it easier to merge multiple versions together (from Khronos, public, and multiple PRs).

An enum entry can also have attributes to describe when the feature is available:

- The `version` attribute corresponds to the
  "[missing before](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#Unified)"
  annotation in the SPIR-V specification.
  The `version` attribute indicates the first SPIR-V version in which the feature exists
  as part of the core specification, without the need to use an `OpExtension` instruction.
  A feature introduced by an extension should always set the `version` attribute to `"None"`.

> - The `extension` attribute, if present, indicates the name of the extension to use

Reviewer: ```suggestion
- The `extension` attribute, if present, indicates the name of an extension to use
```

  with the `OpExtension` instruction when the feature is not yet part of the core SPIR-V specification.
  The `extension` attribute is a list of strings, in case a feature can be exposed through multiple
  extensions.
- If an enum token is only usable when the SPIR-V module declares a capability, then that capability
  must be listed in the `capabilities` attribute for that enum.

>   The `capabiliites` attribute is a list of strings, in case several capabilities can enable a feature.

Reviewer 2: ```suggestion
  The `capabilities` attribute is a list of strings, in case several capabilities can enable a feature.
```

Other tokens introduced by the extension can declare their dependence on one or more of those new capabilities.

For example, this is the entry for the `ShaderViewportIndexLayerEXT` capability:

```
  {
    "enumerant" : "ShaderViewportIndexLayerEXT",
    "value" : 5254,
    "capabilities" : [ "MultiViewport" ],
    "extensions" : [ "SPV_EXT_shader_viewport_index_layer" ],
    "version" : "None"
  },
```

This is one of the tokens enabled by the extension, the `ViewportIndex` `BuiltIn`:

```
  {
    "enumerant" : "ViewportIndex",
    "value" : 10,
    "capabilities" : [ "MultiViewport", "ShaderViewportIndex", "ShaderViewportIndexLayerEXT" ]
  },
```

Note that the `ViewportIndex` token can be enabled by `ShaderViewportIndexLayerEXT`, and in turn
that capability is enabled by the extension.

### Special considerations when adding an instruction

Adding a new instruction is more involved because you need to provide a grammar rule for the operands, and each operand must be assigned the right
operand kind. The rule can be subtle if there are optional, optionally repeating, or optionally repeating pairs of operands. For example, the `OpSwitch` rule is tricky.

> In particular, always use `IdResult` for a result Id operand, and `IdResultType` for a type Id operand. Normal Id operands are just `Id`.

Reviewer: Result IDs are not operands, they are results. Similarly to how we don't call the return value of a function one it's parameters. Similarly for the type ID. This would contradict the way the specification discusses these as well.  SPIR-V does not have normal and non-normal operands.  It is much simpler to always have the first actual operand (the first input to the operation) as the thing we discuss as the first operand.

```suggestion
In particular, always use `IdResult` for a `Result <id>`, and `IdResultType` for a `Type <id>`. The `<id>` operands are just `Id`.
```

I think the specification says this okay too:

> Most instructions create a `Result <id>`, as provided in the `Result <id>` field of the instruction. These `Result <id>`s are then referred to by other instructions through their `<id>` operands.

It would be strange to have to start applying rules about operands also to be about types and results, and then start talking about normal operands to compensate.
